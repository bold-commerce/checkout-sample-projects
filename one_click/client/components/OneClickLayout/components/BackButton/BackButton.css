button.back-btn {
  position: relative;
  display: inline-block;
  border: none;
  margin: 0;
  text-decoration: none;
  background: transparent;
  color: #0069ed;
  font-family: sans-serif;
  font-size: 1rem;
  cursor: pointer;
  text-align: center;
  transition: background 250ms ease-in-out, transform 150ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  padding: 16px 0;
}

button.back-btn::before {
  position: relative;
  border-style: solid;
  border-width: 0.25em 0.25em 0 0;
  content: '';
  display: inline-block;
  height: 0.75em;
  width: 0.75em;
  vertical-align: middle;
  transform: rotate(-135deg);
  margin: 0 5px;
}
