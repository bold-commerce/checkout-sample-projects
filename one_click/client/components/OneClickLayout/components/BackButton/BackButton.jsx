import React from 'react';
import { Link } from "react-router-dom";

const BackButton = () => {
  return (
    <div>
      <Link to="/">
        <button type="button">Back</button>
      </Link>
    </div>
  );
};

export default BackButton;
