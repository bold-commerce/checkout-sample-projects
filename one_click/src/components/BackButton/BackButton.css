button.back-btn {
  position: relative;
  display: inline-block;
  border: none;
  text-decoration: none;
  background: transparent;
  color: #0069ed;
  font-weight: bold;
  font-size: 1rem;
  cursor: pointer;
  text-align: center;
  transition: background 250ms ease-in-out, transform 150ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  margin-top: 16px;
}

button.back-btn::before {
  position: relative;
  border-style: solid;
  border-width: 0.15em 0.15em 0 0;
  content: '';
  display: inline-block;
  height: 0.75em;
  width: 0.75em;
  transform: rotate(-135deg);
  margin: 0 5px;
}
