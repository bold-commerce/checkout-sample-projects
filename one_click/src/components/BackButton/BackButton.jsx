import React from 'react';
import './BackButton.css';
import { useTranslation } from 'react-i18next';

const BackButton = ({ onClick }) => {
  const { t } = useTranslation();
  return (
    <div>
      <button className="back-btn" type="button" onClick={onClick}>{t('back')}</button>
    </div>
  );
};

export default BackButton;
