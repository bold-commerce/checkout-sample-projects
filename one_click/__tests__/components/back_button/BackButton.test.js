import React from "react";
import BackButton from '../../../src/components/OneClickLayout/components/BackButton/BackButton';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router";

describe('BackButton', () => {
    test('renders BackButton component', () => {
        const { asFragment } = render(
            <BackButton />,
            {wrapper: MemoryRouter}
        );
        expect(asFragment()).toMatchSnapshot();
    })
});