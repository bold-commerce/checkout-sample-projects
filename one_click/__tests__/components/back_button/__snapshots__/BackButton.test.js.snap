// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders BackButton component 1`] = `
<DocumentFragment>
  <div>
    <button
      class="back-btn"
      type="button"
    >
      back
    </button>
  </div>
</DocumentFragment>
`;
